app/cdap/components/PushdownConfig/index.tsx

```diff
@@ -167,7 +168,7 @@ export default function PushdownConfig({ value, onValueChange }: IPushdownProps)
       }
     );
     // pass etc
-  }, []);
+  }, [cloudArtifact?.name, cloudArtifact?.scope, cloudArtifact?.version]);
```

Comment:
I think in this case you can just use `cloudArtifact`? When there's a change in one of these values, a new object is created right?

Reply:
Actually yeah, you're right - I was guarding against risk of the object being recreated in PushdownTabContent, but on re-check, it's not. I'll change that